### ISSUE TITLE
PDB Applies wrong type to some functions

### ISSUE THREAD

**Issue by plafosse** [Vector35/binaryninja-api#3639]
**Version and Platform (required):**
- Binary Ninja Version: 3.2

**Bug Description:**

With PDB on right, without PDB on left.
![image](https://user-images.githubusercontent.com/10556585/200324687-3129c51b-5bab-4d14-b824-b979b39ed957.png)

[vcruntime140d.dll.zip](https://github.com/Vector35/binaryninja-api/files/9952086/vcruntime140d.dll.zip)

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Open the enclosed binary
2. Open the enclosed binary with options disabling debug info
3. See memchr

I'm betting this is very similar to a problem we had with DWARF debug info when we were trying to create type libs. Its likely that these functions with wrong type information simply don't have an entry in the PDB because the were written didrectly in asm.

I'm marking this as high impact as it is currently blocking #3388 from being done.

**Comment by plafosse** [comment]
Heres a little script for printing out all the _bad_ functions

```python
for f in bv.hlil_functions():
  ft = f.source_function.function_type
  if ft.confidence == core.max_confidence and ft.return_value == Type.void() and len(ft.parameters) == 0:
    print(f"{f.source_function.start:#x}")
```

**Comment by CouleeApps** [comment]
Can't reproduce on 3.2 dev. Can you upload (public or private) a copy of the pdb it applies when it breaks?

**Comment by plafosse** [comment]
This is the solution I was thinking about too. We ran into the same problem with dwarf.

**Comment by CouleeApps** [comment]
The fix was almost as described, except now it just returns None for the function types since debuginfo doesn't have a way to specify confidence. Will be fixed in dev >= 3.2.3861
